### PR TITLE
Heat conduction action for setting up transfer with THM

### PR DESCRIPTION
**Pull request by andrsd**
## Reason

This simplifies user's life when they want to run heat conduction and THM coupled together using the multi-app system.

## Design

Allow user to just provide the needed parameters and this action will set all the right user objects and transfers on the heat conduction side.

## Impact

Better UX

### CODE REVIEW

File: modules/heat_conduction/doc/content/source/actions/CoupledHeatTransferAction.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Coupled Heat Transfer Action
+
+## Description
+
+The `CoupledHeatTransferAction` action sets up all of the necessary objects for
+doing transfer between heat conduction and thermal hydraulics module.
```
Comment (joshuahansel) [inline modules/heat_conduction/doc/content/source/actions/CoupledHeatTransferAction.md:6]: Mention that this action couples a 1D flow channel in THM to 3D heat conduction.

File: modules/heat_conduction/src/actions/CoupledHeatTransferAction.C [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+CoupledHeatTransferAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription(
+      "Action that controls the creation of all of the necessary objects for "
+      "doing transfer between heat conduction and thermal hydraulics module");
+  params.addRequiredParam<std::vector<BoundaryName>>(
+      "boundary", "The list of boundary IDs for the heat transfer BC");
+  params.addRequiredParam<VariableName>(
+      "T_fluid", "The name fo the variable holding fluid temperature (transferred from THM)");
```
Comment (joshuahansel) [inline modules/heat_conduction/src/actions/CoupledHeatTransferAction.C:27]: Variable to transfer fluid temperature into

File: modules/heat_conduction/src/actions/CoupledHeatTransferAction.C [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+InputParameters
+CoupledHeatTransferAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription(
+      "Action that controls the creation of all of the necessary objects for "
+      "doing transfer between heat conduction and thermal hydraulics module");
+  params.addRequiredParam<std::vector<BoundaryName>>(
+      "boundary", "The list of boundary IDs for the heat transfer BC");
```
Comment (joshuahansel) [inline modules/heat_conduction/src/actions/CoupledHeatTransferAction.C:25]: List of boundary names on the solid side

File: modules/heat_conduction/doc/content/source/actions/CoupledHeatTransferAction.md (hunk 1 above)
Comment (joshuahansel) [reply modules/heat_conduction/doc/content/source/actions/CoupledHeatTransferAction.md:6]: Change the class doxygen and `addClassDescription` accordingly.

File: modules/heat_conduction/src/actions/CoupledHeatTransferAction.C [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  params.addRequiredParam<VariableName>(
+      "T_fluid", "The name fo the variable holding fluid temperature (transferred from THM)");
+  params.addRequiredParam<VariableName>("T", "The name fo the variable holding solid temperature");
```
Comment (joshuahansel) [inline modules/heat_conduction/src/actions/CoupledHeatTransferAction.C:28]: Variable to transfer solid temperature into

File: modules/heat_conduction/src/actions/CoupledHeatTransferAction.C [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  params.addRequiredParam<VariableName>(
+      "htc", "The name fo the variable that hold heat transfer coefficent (transferred from THM)");
```
Comment (joshuahansel) [inline modules/heat_conduction/src/actions/CoupledHeatTransferAction.C:30]: Variable to transfer heat transfer coefficient into

File: modules/heat_conduction/src/actions/CoupledHeatTransferAction.C [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  MooseEnum directions("x y z");
+  params.addRequiredParam<MooseEnum>("direction", directions, "The direction of the layers.");
+  params.addRequiredParam<unsigned int>("num_layers", "The number of layers.");
+  params.addRequiredParam<std::string>("multi_app", "The name of the multi-app.");
+  params.addRequiredParam<UserObjectName>(
+      "T_fluid_user_object",
+      "The name of the user object in the THM multi-app holding T_fluid values.");
+  params.addRequiredParam<UserObjectName>(
+      "Hw_user_object", "The name of the user object in the THM multi-app holding Hw values.");
```
Comment (joshuahansel) [inline modules/heat_conduction/src/actions/CoupledHeatTransferAction.C:39]: Hw -> htc

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: modules/heat_conduction/doc/content/source/actions/CoupledHeatTransferAction.md (hunk 1 above)
Comment (licharlot) [reply modules/heat_conduction/doc/content/source/actions/CoupledHeatTransferAction.md:6]: It also works with 2D heat conduction.

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**